src/ray/core_worker/common.h

```diff
@@ -31,12 +88,21 @@ class TaskArg {
     return TaskArg(std::make_shared<ObjectID>(object_id), nullptr);
   }
 
-  /// Create a pass-by-reference task argument.
+  /// Create a pass-by-value task argument.
   ///
-  /// \param[in] object_id Id of the argument.
+  /// \param[in] value Value of the argument.
```

Comment:
The comment for the `value` argument of these static methods is the same but they have different types. Please fix this.

Also, do we need both of these? Why not just only take a `shared_ptr` and force the caller to make one to make things clear?